lib/newrelic_httprb/instrumentation.rb

```diff
@@ -0,0 +1,33 @@
+DependencyDetection.defer do
+  named :http_rb
+
+  depends_on do
+    defined?(HTTP) && defined?(HTTP::Client)
+  end
+
+  executes do
+    ::NewRelic::Agent.logger.info 'Installing http.rb instrumentation'
+    require 'new_relic/agent/cross_app_tracing'
+    require 'newrelic_http/wrappers'
+  end
+
+  executes do
+    class HTTP::Client
+      def perform_with_newrelic_trace(request, options)
+        wrapped_request = ::NewRelicHTTP::HTTPRequest.new(request)
+
+        ::NewRelic::Agent::CrossAppTracing.tl_trace_http_request(wrapped_request) do
+          # RUBY-1244 Disable further tracing in request to avoid double
+          # counting if connection wasn't started (which calls request again).
```

Comment:
This comment seems to be inspired by https://github.com/newrelic/rpm/blob/master/lib/new_relic/agent/instrumentation/net.rb#L24-L28 .

I'm not sure if this still applies to `http.rb`. The newrelic bug tracker is not public, but their specs that mention RUBY-1244 at <https://github.com/newrelic/rpm/blob/master/test/new_relic/http_client_test_cases.rb> seem to be specific to `NET::HTTP`.

Even if this applies, shouldn't we use the `NewRelic::Agent.disable_all_tracing` that they do in their code?